### PR TITLE
[CWS] make use of device for the mount resolution

### PR DESCRIPTION
**Pull request by safchain**
### What does this PR do?

This PR makes use of the device as a fallback for the mount resolution.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/security/resolvers/path/resolver.go [hunk 1]
```diff
@@ -90,21 +90,27 @@ func (r *Resolver) ResolveFileFieldsPath(e *model.FileFields, pidCtx *model.PIDC
...
-	mountPath, err := r.mountResolver.ResolveMountPath(e.MountID, pidCtx.Pid, ctrCtx.ID)
+	mountPath, err := r.mountResolver.ResolveMountPath(e.MountID, e.Device, pidCtx.Pid, ctrCtx.ID)
 	if err != nil {
 		if _, err := r.mountResolver.IsMountIDValid(e.MountID); errors.Is(err, mount.ErrMountKernelID) {
 			return pathStr, &ErrPathResolutionNotCritical{Err: fmt.Errorf("mount ID(%d) invalid: %w", e.MountID, err)}
 		}
+
+		if e.Device == 265289732 {
+			fmt.Printf("KO ? : %s/%s\n", mountPath, pathStr)
+		}
```
Comment (spikat) [inline pkg/security/resolvers/path/resolver.go:101]: debug to remove? :)

File: pkg/security/resolvers/path/resolver.go [hunk 2]
```diff
@@ -90,21 +90,27 @@ func (r *Resolver) ResolveFileFieldsPath(e *model.FileFields, pidCtx *model.PIDC
...
 	}
 
-	mountPath, err := r.mountResolver.ResolveMountPath(e.MountID, pidCtx.Pid, ctrCtx.ID)
+	mountPath, err := r.mountResolver.ResolveMountPath(e.MountID, e.Device, pidCtx.Pid, ctrCtx.ID)
 	if err != nil {
 		if _, err := r.mountResolver.IsMountIDValid(e.MountID); errors.Is(err, mount.ErrMountKernelID) {
 			return pathStr, &ErrPathResolutionNotCritical{Err: fmt.Errorf("mount ID(%d) invalid: %w", e.MountID, err)}
 		}
+
+		if e.Device == 265289732 {
```
Comment (YoannGh) [inline pkg/security/resolvers/path/resolver.go:99]: to remove

File: pkg/security/resolvers/mount/resolver.go [hunk 3]
```diff
@@ -16,22 +16,23 @@ import (
...
 	"github.com/DataDog/datadog-agent/pkg/security/resolvers/cgroup"
+	cmodel "github.com/DataDog/datadog-agent/pkg/security/resolvers/cgroup/model"
 	"github.com/DataDog/datadog-agent/pkg/security/secl/model"
 	"github.com/DataDog/datadog-agent/pkg/security/utils"
 	"github.com/DataDog/datadog-agent/pkg/util/kernel"
 )
 
 const (
-	numAllowedMountIDsToResolvePerPeriod = 75
+	numAllowedMountIDsToResolvePerPeriod = 1
```
[the remark this answers is not in the document]
Comment (safchain) [reply pkg/security/resolvers/mount/resolver.go]: I have put 5/sec

**Review by YoannGh — APPROVED** [review]

**Review by paulcacheux — APPROVED** [review]

**Review by spikat — APPROVED** [review]

File: pkg/security/probe/field_handlers_linux.go [hunk 4]
```diff
@@ -95,7 +95,7 @@ func (fh *FieldHandlers) ResolveMountPointPath(ev *model.Event, e *model.MountEv
 
 func (fh *FieldHandlers) ResolveMountSourcePath(ev *model.Event, e *model.MountEvent) string {
 	if e.BindSrcMountID != 0 && len(e.MountSourcePath) == 0 {
-		bindSourceMountPath, err := fh.resolvers.MountResolver.ResolveMountPath(e.BindSrcMountID, ev.PIDContext.Pid, ev.ContainerContext.ID)
+		bindSourceMountPath, err := fh.resolvers.MountResolver.ResolveMountPath(e.BindSrcMountID, e.Device, ev.PIDContext.Pid, ev.ContainerContext.ID)
```
Comment (YoannGh) [inline pkg/security/probe/field_handlers_linux.go:98]: not sure this is the correct device number here

**Pull request merged**